src/lib/problem_data.c

```diff
@@ -143,15 +143,15 @@ void problem_data_add_basics(problem_data_t *pd)
                  */
                 if (item->flags & CD_FLAG_BIN)
                     continue;
-                sha1_hash(&sha1ctx, item->content, strlen(item->content));
+                sha1_update(&hash, strlen(item->content), (unsigned char *)item->content);
             }
             g_list_free(list);
 
             /* end hash */
-            char hash_bytes[SHA1_RESULT_LEN];
-            sha1_end(&sha1ctx, hash_bytes);
-            char hash_str[SHA1_RESULT_LEN*2 + 1];
-            bin2hex(hash_str, hash_bytes, SHA1_RESULT_LEN)[0] = '\0';
+            unsigned char hash_bytes[SHA1_DIGEST_SIZE];
+            sha1_digest(&hash, SHA1_DIGEST_SIZE, hash_bytes);
```

Comment:
I’m wondering… Wouldn’t it be better to use `sizeof(hash_bytes)` here and below?

Reply:
Made your dreams come true.